### PR TITLE
chore: license and acknowledgement

### PR DESCRIPTION
**Pull request by technicallyty**
closes: #41 

adds license from vt protobuf as we use their work in the proto methods https://github.com/planetscale/vtprotobuf/blob/main/LICENSE

adds acknowledgements to readme linking to them

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,29 @@
+Copyright (c) 2021, PlanetScale Inc. All rights reserved.
+Copyright (c) 2013, The GoGo Authors. All rights reserved.
+Copyright (c) 2018 The Go Authors. All rights reserved.
```
Comment (aaronc) [inline LICENSE:3]:
```suggestion
Copyright (c) 2018, The Go Authors. All rights reserved.
```

File: README.MD [hunk 2]
```diff
@@ -10,3 +10,8 @@ cd path/to/proto/files
 
 protoc --go-pulsar_out=. --go-pulsar_opt=paths=source_relative --go-pulsar_opt=features=marshal+unmarshal+size -I .
 NAME_OF_FILE.proto
+
+
+## Acknowledgements
+
+[planetscale/vtprotobuf](https://github.com/planetscale/vtprotobuf)
```
Comment (aaronc) [inline README.MD:17]: can we say a bit more about why were acknowledging them and that we're using code?

Comment (technicallyty) [reply README.MD]: hows https://github.com/cosmos/cosmos-proto/pull/42/commits/0ba1e02582b91977731053e8cffbdbc9342de885 ?

**Review by aaronc — APPROVED** [review]

File: README.MD [hunk 3]
```diff
@@ -10,3 +10,8 @@ cd path/to/proto/files
 
 protoc --go-pulsar_out=. --go-pulsar_opt=paths=source_relative --go-pulsar_opt=features=marshal+unmarshal+size -I .
 NAME_OF_FILE.proto
+
+
+## Acknowledgements
+
+Code for marshal, unmarshal, and size implemented by [planetscale/vtprotobuf](https://github.com/planetscale/vtprotobuf) used in our `ProtoMethods` implementation.
```
Comment (aaronc) [inline README.MD:17]:
```suggestion
Code for marshal, unmarshal, and size implemented by [planetscale/vtprotobuf](https://github.com/planetscale/vtprotobuf) was used in our `ProtoMethods` implementation. 
```

**Comment by aaronc** [comment]
@marbar3778 was suggesting using apache. Let's check with him

**Review by ebuchman — COMMENTED** [review]
Ooh, this repo should probably be licensed Apache for consistency. To the extent code under this BSD-3 license is included, a copy of the license needs to be included as well, but should only apply to the relevant files.

File: README.MD (hunk 2 above)
Comment (ebuchman) [reply README.MD]: Would be even better if we could also link to the PR/issue where that code was copied in?

File: LICENSE [hunk 4]
```diff
@@ -0,0 +1,29 @@
+Copyright (c) 2021, PlanetScale Inc. All rights reserved.
+Copyright (c) 2013, The GoGo Authors. All rights reserved.
+Copyright (c) 2018, The Go Authors. All rights reserved.
```
Comment (robert-zaremba) [inline LICENSE:3]: Should we add `Cosmos SDK Team` here?

File: LICENSE [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2021 Regen Network Development, Inc. & All in Bits, Inc.
```
Comment (technicallyty) [inline LICENSE:192]: not sure if this is the right way to put it - feedback?

**Review by technicallyty — COMMENTED** [review]
removed some of the other vt code that was in but we never used. 

not sure where their BSD license would go.. should i just throw it in the directories where their code is used? (generator/ and features/fastreflection) ?

File: LICENSE (hunk 4 above)
Comment (technicallyty) [reply LICENSE]: I have put down the two teams that worked on it (regen and aib) but if cosmos sdk team is preferred or more correct i will change it

**Comment by aaronc** [comment]
is this good to merge or were there more updates to do here?

i think where we left things in the last discussion was maybe that we would do BSD for now for compliance with the other licenses, and then add Apache with guidance from ICF. @marbar3778 do you recall?

**Comment by technicallyty** [comment]
> is this good to merge or were there more updates to do here?
> 
> i think where we left things in the last discussion was maybe that we would do BSD for now for compliance with the other licenses, and then add Apache with guidance from ICF. @marbar3778 do you recall?

iirc the last meeting we wanted apache at the root and then the BSD licenses in the sub-folders where applicable -- thats how this PR is currently, can change if we want to simplify to all BSD for now

**Review by marbar3778 — APPROVED** [review]
LGTM

**Comment by aaronc** [comment]
That's fine. Let's merge this

**Review by aaronc — APPROVED** [review]

**Pull request merged**